pkg/kubectl/proxy/proxy_server.go

```diff
@@ -169,7 +169,13 @@ func makeUpgradeTransport(config *rest.Config) (proxy.UpgradeRequestRoundTripper
 	}
 	rt := utilnet.SetOldTransportDefaults(&http.Transport{
 		TLSClientConfig: tlsConfig,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
```

Comment:
why are we setting `Timeout` and `DualStack` here? shouldn't we just use the defaults?

Reply:
the default dial has no timeout, which can block for minutes. 

I noticed http.DefaultTransport set `Timeout` and `DualStack` 
Maybe can remove `DualStack`, I am not sure. 
```
        // DualStack enables RFC 6555-compliant "Happy Eyeballs"
	// dialing when the network is "tcp" and the host in the
	// address parameter resolves to both IPv4 and IPv6 addresses.
	// This allows a client to tolerate networks where one address
	// family is silently broken.
	DualStack bool
```

Reply:
removed DualStack